src/util/kaldi-table-inl.h

```diff
@@ -298,10 +338,38 @@ template<class Holder>  class SequentialTableReaderScriptImpl:
     }
     std::string line;
     if (getline(script_input_.Stream(), line)) {
-      SplitStringOnFirstSpace(line, &key_, &data_rxfilename_);
-      if (!key_.empty() && !data_rxfilename_.empty()) {
+      std::string data_rxfilename_next, rest;
+      // rest: a temp string which may contain the range specifier.
+      SplitStringOnFirstSpace(line, &key_, &rest);
+      if (!key_.empty() && !rest.empty()) {
```

Comment:
This whole function is very hard to read now that you've added all this parsing code.  It would be better to break it out into a separate function.